packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -30,26 +30,93 @@ export interface FluidAppInsightsLoggerConfig {
 		 */
 		mode: "inclusive" | "exclusive";
 		/**
-		 * Controls the default filtering of log events by their category.
+		 * Controls the filtering of log events.
 		 * Leaving this undefined will be treated as an empty array.
+		 *
+		 * In order for the filters to be valid they must meet the following conditions:
+		 * 1. There must not be any two filters with the same `namespacePattern`.
+		 * 2. All {@link NamespaceFilter} must not have any defined `namespacePatternException` that is not a child of the parent `namespacePattern`
 		 */
 		filters?: TelemetryFilter[];
 	};
 }
 
 /**
- * Object used with an {@link FluidAppInsightsLoggerConfig}
- * to define logic for filtering of telemetry events
+ * A filter used to match against the category of a telemetry event
  *
  * @public
  */
-export interface TelemetryFilter {
+export interface CategoryFilter {
+	/**
+	 * The categories of telemetry events that this filter applies to
+	 */
+	categories: TelemetryEventCategory[];
+}
+
+/**
+ * A filter used to match against the namespaces of a telemetry event
+ *
+ * @public
+ */
+export interface NamespaceFilter {
+	/**
+	 * The namespace pattern to filter telemetry events.
+	 *
+	 * @remarks This will match namespaces that start with the given string. It is not a Regex pattern.
+	 * @example
+	 * "perf:latency" will match any namespace starting with "perf:latency"
+	 */
+	namespacePattern: string;
 	/**
-	 * The category of telemetry event that this filter applies to
+	 * A list of namespace patterns to explicitly exclude from the filter.
+	 *
+	 * @example If you have a namespacePattern of "perf:latency" but want to exclude
```

Comment:
```suggestion
	 * @example
	 If you have a namespacePattern of "perf:latency" but want to exclude
```

Otherwise the text on this line will be treated as the example's title, rather than contents of it.

Reply:
Updated.